src/System.Text.Json/tests/WritableJsonApiTests.cs

```diff
@@ -0,0 +1,667 @@
+                                          .GetProperty("software developers")
+                                          .GetProperty("intern employees");
+            internDevelopers.Add(EmployeesDatabase.GetNextEmployee());
+
+            var students = StudentsDatabase.GetNextStudent().Value as JsonObject;
+            students.GetProperty("grades").GetProperty("maths")["analysis"] = (JsonNumber) 4;
+        }
+
+        /// <summary>
+        /// Modifying Json object key - remove and add
+        /// </summary>
+        [Fact]
+        public static void TestModifyingJsonObjectKeyRemoveAdd()
+        {
+            var manager = EmployeesDatabase.GetManager();
+            var reportingEmployees = manager.GetProperty("reporting employees");
+
+            var softwareDevelopers = reportingEmployees["software developers"];
+            reportingEmployees.Remove("software developers");
+            reportingEmployees.Add("software engineers", softwareDevelopers);
+        }
+
+        /// <summary>
+        /// Modifying Json object key - modify method
+        /// </summary>
+        [Fact]
+        public static void TestModifyingJsonObjectKeyModifyMethod()
+        {
+            var manager = EmployeesDatabase.GetManager();
+            var reportingEmployees = manager.GetProperty("reporting employees");
+
+            reportingEmployees.ModifyPropertyName("software developers", "software engineers");
+        }
+
+        /// <summary>
+        /// Aquiring all values
+        /// </summary>
+        [Fact]
+        public static void TestAquiringAllValues()
+        {
+            var students = new JsonObject(StudentsDatabase.GetTenBestStudents());
+            var studentsWithoutId = students.Values;
+        }
+
+
+        /// <summary>
+        /// Aquiring all primary typed values
+        /// </summary>
+        [Fact]
+        public static void TestAquiringAllPrimaryTypedValues()
+        {
+            var student = StudentsDatabase.GetNextStudent().Value as JsonObject;
+
+            // Checking if student qualifies to get stipend - must have all grades >= 4
+            var allGrades = student.GetProperty("grades").GetAllPrimaryTypedValues();
+            var qualifies = allGrades.Aggregate(true, (qualifies, grade) =>
+            {
+                switch (grade)
+                {
+                    case JsonNumber numberGrade:
+                        qualifies &= numberGrade.GetInt32() >= 4;
+                        break;
+                    case JsonString textGrade:
+                        qualifies &= textGrade.GetString() != "failed";
+                        break;
+                }
+                return qualifies;
+            });
+        }
+
+        /// <summary>
+        /// Aquiring all values of properties with the same name
+        /// </summary>
+        [Fact]
+        public static void TestAquiringAllPropertiesValuesWithSpecificName()
```

Comment:
Do we need/want more advanced methods for looking up properties like `GetAllValuesByPropertyName` and `GetAllPrimaryTypedValues` in `JsonObject` used in this scenario and the one above? They would be helpful, but I'm afraid they are too specialized.

Reply:
I think they're too specialized and not generally applicable to real-world models.